src/unittest/random_graph.cpp

```diff
@@ -0,0 +1,213 @@
+
+namespace vg {
+namespace unittest {
+
+VG randomGraph(int64_t seqSize, int64_t variantLen,
+                    int64_t variantCount){
+    //Create a random graph for a sequence of length seqSize
+    //variantLen is the mean length of a larger variation and variationCount
+    //is the number of variations in the graph
+
+    VG graph;
+    map<size_t, id_t> indexToNode;
+                  //Index of original sequence to node that starts at that index
+
+    string seq(seqSize, 'A');//TODO: The graph is all As
+    Node* n = graph.create_node(seq);
+    indexToNode.insert(make_pair(0, n->id()));
+
+    //Get random number generator for lengths and variation types
+    poisson_distribution<size_t> lengthDistribution(variantLen);
+    uniform_int_distribution<int> variantDistribution(0, 4);
+    uniform_int_distribution<int> indexDistribution(0, seqSize-1);
+    default_random_engine generator(time(NULL));
+
+    auto splitNode = [&] (size_t index) -> pair<Node, Node> {
+        //Split graph at index. Split the node with the original sequence into
+        //the original node and a new node and connect the two
+
+        auto n = --indexToNode.upper_bound(index);//orig node containing pos
+        size_t firstIndex = n->first;           //Index of first node
+        size_t firstNodeID = n->second;         //Node ID of first node
+
+        Node* firstNode = graph.get_node(firstNodeID);
+        string origSeq = firstNode->sequence();
+        if (index > firstIndex) {
+            size_t nodeLength = index - firstIndex;
+
+            firstNode->set_sequence(origSeq.substr(0, nodeLength));//replace seq
+            Node* newNode = graph.create_node(origSeq.substr(nodeLength));
+            indexToNode.insert(make_pair(index, newNode->id()));
+
+
+            //Transfer outgoing edges from fist node to second node
+
+            handle_t startFd = graph.get_handle(firstNodeID, false);
+            handle_t endFd = graph.get_handle(newNode->id(), false);
+
+
+            unordered_set<handle_t> nextHandles;
+            auto addHandle = [&](const handle_t& h) ->bool {
+                nextHandles.insert(h);
+                return true;
+            };
+            graph.follow_edges(startFd, false, addHandle);
+
+            for (handle_t h : nextHandles) {
+                //for each edge from start node, delete and add to new node
+                graph.destroy_edge(startFd, h);
+                graph.create_edge(endFd, h);
+            }
+            graph.create_edge(firstNode, newNode);
+            return make_pair(*firstNode, *newNode);
+        } else {
+
+            auto n = --indexToNode.lower_bound(index);
+            Node* prevNode = graph.get_node(n->second);
+            return make_pair(*prevNode, *firstNode);
+        }
+
+    };
+
+    for (int j = 0; j < variantCount; j++) {
+        //add variants
+        int startIndex = indexDistribution(generator);
+        int variationType = variantDistribution(generator);
```

Comment:
Not super important, but a nice stylistic option for cases like this is to make a local enum. It could be something like 
```
enum VariationType {SNP = 0, POINT_INDEL = 1, STRUCTURAL_INDEL = 2, CNV = 3, INVERSION = 4};
```
You can cast straight from `int`, so it looks like this:
```
VariationType variationType = variantDistribution(generator);
if (variationType == SNP) {
   ...
}
```
Much more readable!